docs/howto/composite-primary-key.txt

```diff
@@ -0,0 +1,171 @@
+=====================================
+How to define a composite primary key
+=====================================
+
+.. versionadded:: 5.2
+
+In Django, each model has a primary key. By default, this primary key consists
+of a single field.
+
+In most cases, a single primary key should suffice. In database design,
+however, defining a primary key consisting of multiple fields is sometimes
+necessary.
+
+To define a composite primary key, set the model's ``pk`` to an instance of
+``CompositePrimaryKey``:
+
+.. code-block:: pycon
+
+    class Tenant(models.Model):
+        pass
+
+    class User(models.Model):
+        pk = models.CompositePrimaryKey("tenant_id", "id")
+        tenant = models.ForeignKey(Tenant, on_delete=models.CASCADE)
+        id = models.IntegerField()
+
+This will instruct Django to create a composite primary key
+(``PRIMARY KEY (tenant_id, id)``) when creating the table.
+
+It's possible to define a composite primary key consisting of a single field.
+This is _almost_ the same as setting ``primary_key=True`` on the field.
+
+.. code-block:: pycon
+
+    class Foo(models.Model):
+        pk = models.CompositePrimaryKey("id")
+        id = models.IntegerField()
```

Comment:
Or we could prevent them from doing this? I don't really have any arguments for or against doing this.

The only difference is `pk` will be a tuple, and the primary key will be created as a table-level constraint instead of a column-level constraint when creating the table.

Reply:
I see 🤔 maybe let's not support and then if people give reasons why they want it, we can add support. It's probably more likely to be a mistake and then they can't migrate after

Reply:
I added an error for this.